### DISCUSSION TITLE
how do I change the value of a GPIO?

### DISCUSSION THREAD

**Discussion by arrow53** [OE4T/tegra-demo-distro#310] [Q&A]
I'm working with a Jetson Orin Nano and scarthgap.

Previously i've used `/sys/class/gpio` to set GPIO values. But, that seems to be missing.

Can anyone provide some general guidance on setting `GPIO02` and `GPIO03` which are `GPIO0_PP.06` and `GPIO3_PCC.00` respectively.

I'm pretty unclear to where to start.
thanks!

**Comment by dwalkes** [comment]
@arrow53 I'd start by mining the content in https://github.com/search?q=org%3AOE4T+gpio&type=discussions and possibly also referencing the wiki content in https://github.com/search?q=org%3AOE4T+gpio&type=wikis.  We do have a page on GPIO control at https://github.com/OE4T/meta-tegra/wiki/Controlling-Jetson-TX2-pin-states but I think it's out of date and in need of a refresh/update.

**Reply by madisongh** [reply]
Also, some Google searching for "GPIO sysfs vs libgpiod" will also give you some pointers to info on the years-long transition away from the old sysfs interface for manipulating GPIOs, too. You should still be able to add `CONFIG_GPIO_SYSFS=y` to your kernel config to enable the old interface, though.

**Reply by arrow53** [reply]
thanks guys. the `CONFIG_GPIO_SYSFS` is nice

**Comment by arrow53** [comment]
For posterity in case anyone else runs across this discussion, this is what I did

1. Grab the pinmux spreadsheet and configure the pins the way you need then generate the new files https://developer.nvidia.com/downloads/jetson-orin-nx-and-orin-nano-series-pinmux-config-template. 
2. This will give you three new dtsi files. You need to match up what these are with your machine and meld them to get the changes you need from the machine. The relevant recipe is `tegra-bootfiles`
3. Build in these recipes `libgpiod libgpiod-tools libgpiod-dev`
4. Back at the command line run `gpioinfo` and grep on your gpio you want. For my case I wanted `GGPIO3_PCC.00`
5. Take the controller name (0 or 1 for me) and the line and you should now be able to `gpioset -c 1 12=1` to set. where the c is the controller number and 12 is the line number.

Good reference https://docs.nvidia.com/jetson/archives/r35.3.1/DeveloperGuide/text/HR/JetsonModuleAdaptationAndBringUp/JetsonOrinNxNanoSeries.html#generating-the-pinmux-dtsi-files

**Reply by dwalkes** [reply]
Thanks for sharing! I've copied to https://github.com/OE4T/meta-tegra/wiki/Controlling-Jetson-pin-states so we can find this link in wiki searches going forward.
